src/community/wps-download/src/test/java/org/geoserver/wps/gs/download/DownloadProcessTest.java

```diff
@@ -1379,6 +1398,102 @@ public void testDownloadScaledRasterPreservingDatatype() throws Exception {
         }
     }
 
+    /**
+     * Test download of raster data. The source is an ImageMosaic with Heterogeneous CRS. Sending a
+     * request with a TargetCRS matching one of the underlying CRS of that mosaic should result in
+     * minimal reprocessing on granules with that CRS as native.
+     */
+    @Test
+    public void testDownloadGranuleHeterogeneousCRS() throws Exception {
```

Comment:
Tests only the happy path. What about a case where the bbox is not catching the target CRS? Or having a target width and height set?